src/client.rs

```diff
@@ -879,6 +883,26 @@ impl Client {
                             continue;
                         }
 
+                        //check if subscription has set limits for slow consumers
+                        if let Some(pending_messages_limit) = subscription.pending_messages_limit {
+                            if pending_messages_limit <= subscription.messages.len() {
+                                connector.get_options().error_callback.call(
```

Comment:
Maybe a callback specific for this state detection? `slow_consumer_callback`? That way you can pass the subscription in.

Reply:
Discussed offline, Go doesn't so lets keep it as is until the need is proven.